### ISSUE TITLE
Storing star rating in Firebase

### ISSUE THREAD

**Issue by alreadyHub** [evgenyneu/Cosmos#128]
Please consider submitting the following information (if relevant):

* Library setup method: Cocoapods
* Xcode version. 10

Is it possible to store the result of the Cosmos Review in a Firebase database so I can retrieve and display on a different screen? 

I am familiar with Firebase but I'm not sure what data type Cosmos is ?

**Comment by evgenyneu** [comment]
> Is it possible to store the result of the Cosmos Review in a Firebase database so I can retrieve and display on a different screen?

Yes, one can supply a closure to `didFinishTouchingCosmos` to save the rating in any database, including Firebase. The closure will be called when user changes the rating and lifts a finger from the screen. For example:

```Swift
cosmosView.didFinishTouchingCosmos = { rating in 
   // Save the rating in the database
} 
```

> I am familiar with Firebase but I'm not sure what data type Cosmos is ?

The `rating` variable is of `Double` type, it is a number from 0 to 5 (when there are five total stars).

The demo app shows how to save and load ratings to multiple Cosmos views to a local storage ([Demo/PerformanceTableViewController.swift](https://github.com/evgenyneu/Cosmos/blob/master/Demo/PerformanceTableViewController.swift).

**Comment by alreadyHub** [comment]
thanks a bunch, @evgenyneu . i appreciate the help. i might be back with more questions but I'll be sure to do my due diligence before asking.

**Comment by alreadyHub** [comment]
what are the "override func" for? Why not just put those functions with class above?

**Comment by evgenyneu** [comment]
Sorry I don't understand what you mean. Can you explain the question?

**Comment by patskid315** [comment]
is cosmosView still open for Q&A?

**Comment by evgenyneu** [comment]
Hi, @patskid315, what do you mean sorry?

**Comment by patskid315** [comment]
I am using cosmosView to store ratings but I am having a hard time understanding how to retrieve the information from Firebase, and store the value as stars on my viewController. I asked if it was still open to see if someone who worked on it knows how to accomplish this!

**Comment by evgenyneu** [comment]
I have never used Firebase myself, but of course everyone is welcome to comment here, since this is an open source project.
